### ISSUE TITLE
Persisting index.yaml in the storage

### ISSUE THREAD

**Issue by alekc** [helm/chartmuseum#517]
Judging from the docs, it looks like at the moment chart museum only supports `--gen-index` which can be used for a static generation of index.yaml (but since it happens in stdout, operator would need to capture it and upload to the storage manually which can be problematic). 

Is there any reason for which chartmuseum doesn't support saving `index.yaml` at the same time of `index-cache.yaml`?

An example scenario where such thing would be very useful is for example with Backblaze which permits to have a free egress fronted by Cloudflare (so even if chartmuseum is not running, charts are still available online). 

A possible configuration would be the following:

- Chartmuseum running on chartmuesum.example.com and deals with upload authentication, persistence, and generation of index.yaml
- When chartmuseum detects a change in charts / receives new chart, it will generate and save a index.yaml file using `--chart-url` as a prefix
- If needed files can be accessed through either the Cloudflare (charts.example.com) or through the chartmuseum as it happens currently.

**Comment by scbizu** [comment]
I agree with your opinion , but currently CM is the server-based arch , not designed perfectly for no-server(serverless) arch . The mechanism in severless should be separate from the server-based chartmuseum in my opinion  (set up a new  `serverless-cm` repo is a work-out solution) . Anyway, thanks for your idea 👍
